cli/context/store/io_utils.go

```diff
@@ -0,0 +1,67 @@
+package store
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// limitedReader a wrapper on io.Reader for error handling of ReadAll
+type limitedReader struct {
+	r     io.Reader
+	limit int64
+	n     int64
+	err   error
+}
+
+// limitReaderWithErrorHandling will result in a limited reader with defined byte limit.
+// This basically extends io.LimitReader with proper errors as io.LimitReader only errors with EOF.
+func limitReaderWithErrorHandling(r io.Reader, l int64) io.Reader {
+	return &limitedReader{r: r, limit: l, n: l}
+}
+
+// LimitedReadAll will read all content of a limited reader.
+// Should be called with a Reader that's gathered from limitReaderWithErrorHandling
+// Safer than using regular ioutil.ReadAll() that may result in issues on very big files.
+func LimitedReadAll(r io.Reader, limit int64) ([]byte, error) {
+	r = limitReaderWithErrorHandling(r, limit)
+	return ioutil.ReadAll(r)
+}
+
+// ReadExceedsLimitError thrown error type when read exceeds expected limit on reader
+type ReadExceedsLimitError struct {
```

Comment:
Looks like this error doesn't have to be exported (there's no special handling for this error); if it's only to be used for the unit test, we could keep it simple and just return `return fmt.Errorf()` or `return error.Errorf()` and just check for the error string?

Reply:
Changed it back to being an error string. My original thinking was around having strings and possible mistypes, but since it's only in two places and not public properties, it was an overkill.